Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -621,6 +623,14 @@ public void onInventoryClickEvent(final InventoryClickEvent event) {
                 if (user.isInvSee() && (!user.isAuthorized("essentials.invsee.modify") || invOwner.isAuthorized("essentials.invsee.preventmodify") || !invOwner.getBase().isOnline())) {
                     event.setCancelled(true);
                     refreshPlayer = user.getBase();
+                } else if (ess.getSettings().isDirectHatAllowed() && ess.getUser(event.getWhoClicked()).isAuthorized("essentials.hat") && event.getSlot() == 38 && event.getClick() == ClickType.LEFT) {
```

Comment:
I'd front load it with minimal-effort checks like the slot int and click type, ending in the isAuthorized call.